keps/sig-node/3085-pod-conditions-for-starting-completition-of-sandbox-creation/README.md

```diff
@@ -1,1618 +1,1618 @@
+ the CRI container runtime. The `PodHasNetwork` condition will mark an important
+ milestone in the pod's lifecycle similar to `ContainersReady` and the overall
+ `Ready` conditions in pod status today. In the future, a dedicated
+ `PodHasVolumes` condition may be considered to mark successful mounting and
+ setup of all volumes in the pod spec. An alternate name like `SandboxReady` is
+ avoided since Kubernetes does not directly surface low level sandbox related
+ concepts to all users.
+ 
+ ## Motivation
+ 
+ <!--
+ This section is for explicitly listing the motivation, goals, and non-goals of
+ this KEP.  Describe why the change is important and the benefits to users. The
+ motivation section can optionally provide links to [experience reports] to
+ demonstrate the interest in a KEP within the wider Kubernetes community.
+ 
+ [experience reports]: https://github.com/golang/go/wiki/ExperienceReports
+ -->
+ 
+ Today, the scheduler surfaces a specific pod condition: `PodScheduled` that
+ clearly identifies whether a pod got scheduled by the scheduler and when
+ scheduling completed. However, no specific conditions around initialization of
+ successfully scheduled pods from the perspective of completion of pod sandbox
+ creation (marked by the presence of a pod sandbox with networking configured) is
+ surfaced to cluster administrators in a scoped and consumable fashion.
+ 
+ There is an existing pod condition: `Initialized` that tracks execution of init
+ containers. For pods without init containers, the `Initialized` condition is set
+ when the Kubelet starts to process a pod before any sandbox creation activities
+ start. For pods with init containers, the `Initialized` condition is set when
+ init containers have been pulled and executed to completion. Therefore, the
+ existing `Initialized` condition is insufficient and inaccurate for tracking
+ completion of sandbox creation of all pods in a cluster. This distinction
+ becomes especially relevant in multi-tenant clusters where individual tenants
+ own the pod specs (including the set of init containers) while the cluster
+ administrators are in charge of storage plugins, networking plugins and
+ container runtime handlers.
+ 
+ Conclusion of the creation of the pod sandbox is marked by the presence of a
+ sandbox with networking configured. A new dedicated condition marking this -
+ `PodHasNetwork` - will benefit cluster operators (especially of multi-tenant
+ clusters) who are responsible for configuration and operational aspects of the
+ various components that play a role in pod sandbox creation: CSI plugins, CRI
+ runtime and associated runtime handlers, CNI plugins, etc. The duration between
+ `lastTransitionTime` field of the `PodHasNetwork` condition (with `status` set
+ to `true` for a pod for the first time) and the existing `PodScheduled`
+ condition will allow metrics collection services to compute total latency of all
```

Comment:
I'm concerned about using transition times of different conditions to measure duration of operations. For phases that must be performed in series and that have a guaranteed stable ordering, this would seem okay, but it seems like network configuration could happen in parallel with other steps, maybe as an optimization in the future. Also, in the future, if there needs to be some step after network it looks like it would mess up metrics collection (i.e. if X happens after network configuration, you could end up measuring the latency of network configuration + latency of X).